### ISSUE TITLE
报错RuntimeError: Error building extension 'fused_adam'

### ISSUE THREAD

**Issue by zbxx-423** [THUDM/VisualGLM-6B#292]
![Snipaste_2023-10-19_15-41-33](https://github.com/THUDM/VisualGLM-6B/assets/147784112/ad95e8f9-7951-4634-829f-87001c81187a)
这是cuda版本算力问题吗

**Comment by 1049451037** [comment]
参考这里：https://github.com/THUDM/VisualGLM-6B/issues/125#issuecomment-1630407747

**Comment by LijunRio** [comment]
我将cuda的版本改为11.7之后能解决这个问题，主要是deepspeed的问题

**Comment by chenchen333-dev** [comment]
> 我将cuda的版本改为11.7之后能解决这个问题，主要是deepspeed的问题

是deepspeed版本的问题吗，什么版本呢

**Comment by ToStupid** [comment]
怎么解决呢？

**Comment by Gae-Zhang** [comment]
@zzyzeyuan  Have you fixed it?

**Comment by zzyzeyuan** [comment]
> @zzyzeyuan Have you fixed it?

Yes, see [this page](https://github.com/databrickslabs/dolly/issues/119).

The error may be caused by the wrong installation of `deepspeed`, and you just need to clone the repo of `deepspeed` and install it manually like:
```
git clone https://github.com/microsoft/DeepSpeed.git
cd DeepSpeed
DS_BUILD_FUSED_ADAM=1 pip3 install .
```

**Comment by chenchen333-dev** [comment]
> > 这不是算力的问题，是nvcc压根没安装……或者没加到环境变量里
> 
> nvcc放到他指定的目录里解决掉之后 还是报RuntimeError: Error building extension 'fused_adam' 这个错误 我用的是两张3090 报错之前看显卡使用都是100%

要配置cuda环境路径，这个问题是因为找不到cuda的位置，执行vi ~/.bashrc打开编辑脚本

#添加环境变量，XX为版本号
export CUDA_HOME=/usr/local/cuda-版本号
export PATH=$CUDA_HOME/bin:$PATH
export LD_LIBRARY_PATH=$CUDA_HOME/lib64:$LD_LIBRARY_PATH

**Comment by Gae-Zhang** [comment]
> > @zzyzeyuan Have you fixed it?
> 
> Yes, see [this page](https://github.com/databrickslabs/dolly/issues/119).
> 
> The error may be caused by the wrong installation of `deepspeed`, and you just need to clone the repo of `deepspeed` and install it manually like:
> 
> ```
> git clone https://github.com/microsoft/DeepSpeed.git
> cd DeepSpeed
> DS_BUILD_FUSED_ADAM=1 pip3 install .
> ```

非常感谢您，您的回答解决了困扰我的问题，祝您科研顺利！！

**Comment by Gae-Zhang** [comment]
> > @zzyzeyuan Have you fixed it?
> 
> Yes, see [this page](https://github.com/databrickslabs/dolly/issues/119).
> 
> The error may be caused by the wrong installation of `deepspeed`, and you just need to clone the repo of `deepspeed` and install it manually like:
> 
> ```
> git clone https://github.com/microsoft/DeepSpeed.git
> cd DeepSpeed
> DS_BUILD_FUSED_ADAM=1 pip3 install .
> ```

Thank you so much!! 祝您科研顺利！

**Comment by Acrzem** [comment]
> > @zzyzeyuan Have you fixed it?
> 
> Yes, see [this page](https://github.com/databrickslabs/dolly/issues/119).
> 
> The error may be caused by the wrong installation of `deepspeed`, and you just need to clone the repo of `deepspeed` and install it manually like:
> 
> ```
> git clone https://github.com/microsoft/DeepSpeed.git
> cd DeepSpeed
> DS_BUILD_FUSED_ADAM=1 pip3 install .
> ```

非常感谢，先更新gcc版本再按照这个方法安装解决了
